tiledb/api/src/datatype/logical.rs

```diff
@@ -0,0 +1,143 @@
+use crate::datatype::physical::PhysicalType;
+use crate::datatype::Datatype;
+use crate::private::sealed;
+
+/// Trait which provides statically-typed attributes for a TileDB `Datatype`
+/// for use with generics.
+pub trait LogicalType: crate::private::Sealed {
```

Comment:
~~How is this possible? That blog post I linked in a different comment uses this specifically as a "you can't do this" so I'm surprised it compiles. I'm assuming that I'm missing a privacy specifier else where, but if so then we might want to tag this with a comment about the how part.~~

Derps again. Its using the first pattern after the bad pattern where the Sealed trait is public inside a private module. Which... huh. Perhaps that's a thing? Actually, does that mean our use of `pub mod` everywhere is probably wrong? Something like, everything module should be private and then we go through and re-export public symbols with clear intention? I did mention the other day we should clean up some of our exports and that feels like the sort of approach that'd make things clear. Obviously that's not part of this story. Just something to think about.